presto-raptor/src/main/java/com/facebook/presto/raptor/storage/Row.java

```diff
@@ -60,32 +66,77 @@ public static Row extractRow(Page page, int position, List<Type> types)
         for (int channel = 0; channel < page.getChannelCount(); channel++) {
             Block block = page.getBlock(channel);
             Type type = types.get(channel);
+            int size;
             if (block.isNull(position)) {
-                rowBuilder.add(null, SIZE_OF_BYTE);
+                size = SIZE_OF_BYTE;
             }
             else if (type.getJavaType() == boolean.class) {
-                rowBuilder.add(type.getBoolean(block, position), SIZE_OF_BYTE);
+                size = SIZE_OF_BYTE;
             }
             else if (type.getJavaType() == long.class) {
-                rowBuilder.add(type.getLong(block, position), SIZE_OF_LONG);
+                size = SIZE_OF_LONG;
             }
             else if (type.getJavaType() == double.class) {
-                rowBuilder.add(type.getDouble(block, position), SIZE_OF_DOUBLE);
+                size = SIZE_OF_DOUBLE;
             }
             else if (type.getJavaType() == Slice.class) {
-                Slice slice = type.getSlice(block, position);
-                Object value = type.equals(VARCHAR) ? slice.toStringUtf8() : slice.getBytes();
-                rowBuilder.add(value, slice.length());
+                size = type.getSlice(block, position).length();
```

Comment:
Could this be switched so the value object is extracted first, then the object to get the size?